src/sentry/seer/code_review/webhook_task.py

```diff
@@ -0,0 +1,130 @@
+
+    should_record_latency = True
+
+    try:
+        status = make_seer_request(original_run_id)
+    except HTTPError as e:
+        # Catches all urllib3 errors: TimeoutError, MaxRetryError, NewConnectionError,
+        # SSLError, ProxyError, ProtocolError, ResponseError, DecodeError, etc.
+        status = e.__class__.__name__
+        metrics.incr(f"{PREFIX}.outcome", tags={"status": f"error_{status}"})
+        if self.request.retries >= MAX_RETRIES - 1:
+            logger.exception("%s.error", PREFIX, extra=context)
+        else:
+            should_record_latency = False
+        raise  # Re-raise to trigger task retry (error listed in on= parameter)
+    except Exception as e:
+        # Unexpected errors (JSON parsing, config issues, etc.)
+        # These are not retryable and indicate a programming or configuration error
+        status = f"unexpected_error_{e.__class__.__name__}"
+        logger.exception("%s.unexpected_error", PREFIX, extra=context)
+        raise  # Task will fail (no retry for unexpected errors, not listed in on= parameter)
+    finally:
+        metrics.incr(f"{PREFIX}.outcome", tags={"status": status})
+        if should_record_latency:
+            record_latency(status, enqueued_at_str)
+
+
+def make_seer_request(original_run_id: str) -> str:
+    payload = {"original_run_id": original_run_id}
+    status = "failure"
+    response = make_signed_seer_api_request(
+        connection_pool=connection_from_url(settings.SEER_AUTOFIX_URL),
+        path=SEER_PR_REVIEW_RERUN_PATH,
+        body=orjson.dumps(payload),
+    )
+    # Retry on server errors (5xx) and rate limits (429), but not client errors (4xx)
+    if response.status >= 500 or response.status == 429:
+        # Server error or rate limit - transient, should retry
+        raise HTTPError(f"Seer returned retryable status {response.status}")
+    elif response.status >= 400:
+        # Client error (4xx except 429) - permanent, don't retry
+        status = f"client_error_{response.status}"
+    else:
+        status = "success"
+
+    return status
+
+
+def record_latency(status: str, enqueued_at_str: str) -> None:
+    latency_ms = calculate_latency(enqueued_at_str)
+    if latency_ms > 0:
+        metrics.timing(f"{PREFIX}.e2e_latency", latency_ms, tags={"status": status})
+
+
+def calculate_latency(enqueued_at_str: str) -> int:
+    """Calculate the latency between the enqueued_at timestamp and the current time."""
+    try:
+        enqueued_at = datetime.fromisoformat(enqueued_at_str)
+        processing_started_at = datetime.now(timezone.utc)
+        return int((processing_started_at - enqueued_at).total_seconds() * 1000)
+    except (ValueError, TypeError) as e:
+        # Log but don't fail the task if timestamp parsing fails
+        logger.warning(
+            "%s.invalid_timestamp",
+            PREFIX,
+            extra={"enqueued_at": enqueued_at_str, "error": str(e)},
+        )
```

Comment:
i'm kind of curious if this would ever actually fail. do we really need to try/except this?

Reply:
and if so, logging the error doesn't seem ideal, could we use sentry issues instead?